app/kuma-dp/pkg/dataplane/envoy/envoy.go

```diff
@@ -96,7 +103,11 @@ func lookupEnvoyPath(configuredPath string) (string, error) {
 
 func (e *Envoy) Start(stop <-chan struct{}) error {
 	runLog.Info("generating bootstrap configuration")
-	bootstrapConfig, err := e.opts.Generator(e.opts.Config.ControlPlane.URL, e.opts.Config, e.opts.Dataplane)
+	envoyVersion, err := e.version()
+	if err != nil {
+		return errors.Errorf("Failed to get Envoy version. %v", err)
+	}
+	bootstrapConfig, err := e.opts.Generator(e.opts.Config.ControlPlane.URL, e.opts.Config, e.opts.Dataplane, *envoyVersion)
```

Comment:
```
	envoyVersion, err := e.version()
	if err != nil {
		return errors.Wrap(err, "failed to get Envoy version)
	}
	runLog.Info("fetched Envoy version", "version", envoyVersion)
	runLog.Info("generating bootstrap configuration")
	bootstrapConfig, err := e.opts.Generator(e.opts.Config.ControlPlane.URL, e.opts.Config, e.opts.Dataplane, *envoyVersion)

```
1) Slightly different order of logs
2) I think it's useful to add a log of the Envoy version
3) Errors in Go are lowercase
4) Use `errors.Wrap` when you are adding context to errors

Reply:
Thank you. I fixed it.